pkg/services/oauth/oauth.go

```diff
@@ -114,9 +123,32 @@ func (s *OAuthService) ParseAndValidateJwtBearerToken(acString string) (*service
 	return nil, err
 }
 
+func getCertificateFromHeaders(token *jwt.Token) (*x509.Certificate, error) {
+	h, ok := token.Header["x5c"]
+	if !ok {
+		return nil, ErrMissingCertificate
+	}
+	i, ok := h.([]interface{})
+	if !ok {
+		return nil, ErrInvalidX5cHeader
+	}
+	if len(i) != 1 {
+		return nil, ErrInvalidX5cHeader
+	}
+	c, ok := i[0].(string)
+	if !ok {
+		return nil, ErrInvalidX5cHeader
+	}
+	bytes, err := base64.StdEncoding.DecodeString(c)
+	if err != nil {
+		return nil, fmt.Errorf("invalid x5c certificate encoding: %w", err)
```

Comment:
can't just wrap err in ErrInvalidX5cHeader?

Reply:
wrapped the underlying error with the ErrInvalidX5cHeader msg